components/post_view/message_attachments/action_menu/action_menu.tsx

```diff
@@ -23,7 +25,19 @@ type Provider = GenericUserProvider | GenericChannelProvider | MenuActionProvide
 
 type Selected = Option | UserProfile | Channel;
 
-export type Props = OwnProps & PropsFromRedux;
+export type Props = {
+    postId: string;
+    action: PostAction;
+    disabled?: boolean;
+    selected?: Selected;
+    actions: {
```

Comment:
Why is the change in interface/structure of the props needed? (I mean vs `OwnProps & PropsFromRedux` where actions were accessed via `props.{action}`?

Reply:
Yes i agree, but the way it was defined before:-
Our component was not getting the type provided by the its index.ts file